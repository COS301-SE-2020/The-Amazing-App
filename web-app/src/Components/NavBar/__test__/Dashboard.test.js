const { Item } = require("semantic-ui-react")
import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme,{mount ,shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Modal from '../../DashBoard/SideNavigation/MyGames/MyGamesModal'
import { BrowserRouter as Router } from "react-router-dom";
import { Provider } from "react-redux";
import { createStore, applyMiddleware } from 'redux';

Enzyme.configure({ adapter: new Adapter() });
const store = createStore(() => [], {}, applyMiddleware());

describe('dashboard MyGame component test',()=>
{
    it('it should render',()=>{
        const component = renderer.create(<Modal />);

        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });


    it('should render my games',()=>{
        const button = shallow(<Modal />);
        expect(button.find('MyGames').length).toBe(1);
        //expect(button_.length).toBe(1);
    });

    it('should not render any Divs', () => {
        const calculator = shallow(<Modal />);  
        // expect(links.length).toBe(0);
     //   const length = 0;
       // expect(form.find('input').length).toBe(2);
        expect(calculator.find('div').length).toBe(0);
       
        //expect(form.find('button').length).toBe(1);
       // expect(form.find('p.result').length).toBe(1);
      });    
    /*  it('should render required Divs', () => {
        const calculator = shallow(<MDBFooter />);  
        // expect(links.length).toBe(0);
     //   const length = 0;
       // expect(form.find('input').length).toBe(2);
       const internal = calculator.find('div.footer-copyright')
        expect(internal.find('MDBContainer').length).toBe(1);
       
        //expect(form.find('button').length).toBe(1);
       // expect(form.find('p.result').length).toBe(1);
      });    */


})