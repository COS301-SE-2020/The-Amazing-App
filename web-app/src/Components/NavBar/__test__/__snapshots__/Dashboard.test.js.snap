// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dashboard MyGame component test it should render 1`] = `
<button
  className="ui medium button"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "backgroundColor": "#2A9D8F",
      "color": "white",
      "fontSize": 13,
      "marginLeft": 10,
      "marginTop": 12,
      "width": 135,
    }
  }
>
  <i
    className="trophy icon"
  />
  My Games
</button>
`;
